### ISSUE TITLE
`invalid-argument-type` false positive: ParamSpec not propagated through composed higher-order functions

### ISSUE THREAD

**Issue by eitan-protego** [astral-sh/ty#3691]
### Summary

ParamSpec not resolved when passing result of one generic callable wrapper to another

## Description

When a function with `ParamSpec` parameters is passed through two nested higher-order generic functions — e.g. `soonify(asyncify(fn))` — ty fails to propagate the `ParamSpec` from the inner function through to the outer wrapper. Instead, the outer function sees `P` as unresolved and rejects the concrete function argument.

This blocks a pattern in async library `asyncer` where synchronous functions are lifted to async and then wrapped for concurrent execution.

### Repro ([Playground](https://play.ty.dev/511e5af8-51ba-4eca-9a60-a297754fe1e7))

```python
from typing import ParamSpec
from collections.abc import Callable, Awaitable, Coroutine

P = ParamSpec("P")


def asyncify(fn: Callable[P, int]) -> Callable[P, Coroutine[None, None, int]]:
    async def wrapper(*args: P.args, **kwargs: P.kwargs) -> int:
        return fn(*args, **kwargs)
    return wrapper


class Soon[T]:
    value: T


def soonify(fn: Callable[P, Awaitable[int]]) -> Callable[P, int]:
    def wrapper(*args: P.args, **kwargs: P.kwargs) -> int:
        raise NotImplementedError
    return wrapper


def read_items(db: str, limit: int = 10) -> int:
    return limit


# Direct asyncify — works:
async_read = asyncify(read_items)

# Composed soonify(asyncify(...)) — should also work:
composed = soonify(asyncify(read_items))
result = composed("mydb", limit=5)
```

### Expected behavior

`asyncify(read_items)` returns `Callable[[str, int], Coroutine[None, None, int]]` — its `P` is bound to `(db: str, limit: int)`. Passing this to `soonify` should bind `soonify`'s `P` to the same parameters. The composed call `soonify(asyncify(read_items))("mydb", limit=5)` should type-check.

### Actual behavior

```
error[invalid-argument-type]: Argument to function `asyncify` is incorrect:
  Expected `(**P@soonify) -> int`, found `def read_items(db: str, limit: int = 10) -> int`
```

ty fails to resolve `P` through the composition and shows `P@soonify` (unresolved) instead of the concrete parameter spec.

### Other type checkers

| Checker | Result |
|---|---|
| pyright 1.1.400 | pass |
| basedpyright 1.39.2 | pass |
| mypy 1.20.2 | pass |
| pyrefly 0.64.1 | pass |
| **ty 0.0.44** | **error** |

### Related

- #3555 — Support `ParamSpec` in constraint set implementation (tracking issue for the underlying solver limitation)

### Version

ty 0.0.44 (f5523e256 2026-06-04)

**Comment by dhruvmanila** [comment]
Thanks for the report! Yeah, I think this should be solved by #3555, going to mark this as a sub-issue to test this kind of examples once the issue is fixed.

**Comment by charliermarsh** [comment]
We now preserve `(db: str, limit: int = 10) -> int`.
